### PR TITLE
Added feature: usage of systemd-gpt-auto-generator

### PR DESCRIPTION
**Pull request by heroldtech**

### CODE REVIEW

File: alis-commons.conf [hunk 1]
```diff
@@ -16,7 +16,7 @@ PARTITION_BOOT_NUMBER=""
 PARTITION_ROOT_NUMBER=""
 DEVICE_ROOT=""
 DEVICE_LVM=""
-LUKS_DEVICE_NAME="cryptroot"
+LUKS_DEVICE_NAME="root"
```
Comment (picodotdev) [inline alis-commons.conf:19]: What is the reason for this change?

File: alis.sh [hunk 2]
```diff
@@ -419,7 +424,7 @@ function partition() {
         if [ "$BIOS_TYPE" == "uefi" ]; then
             parted -s "$DEVICE" "$PARTITION_PARTED_UEFI"
             if [ -n "$LUKS_PASSWORD" ]; then
-                sgdisk -t="$PARTITION_ROOT_NUMBER":8309 "$DEVICE"
+                sgdisk -t="$PARTITION_ROOT_NUMBER":8304 "$DEVICE"
```
Comment (picodotdev) [inline alis.sh:427]: What is the reason for this change?

File: alis.sh [hunk 3]
```diff
@@ -747,7 +756,7 @@ function mkinitcpio_configuration() {
     if [ "$LVM" == "true" ]; then
         HOOKS=${HOOKS//!lvm2/lvm2}
     fi
-    if [ "$BOOTLOADER" == "systemd" ]; then
+    if [ "$BOOTLOADER" == "systemd" ] || [ "$GPT_AUTOMOUNT" == "true" ]; then
```
Comment (picodotdev) [inline alis.sh:759]: Is necessary add this or condition?

**Review by picodotdev — APPROVED** [review]

**Comment by picodotdev** [comment]
For reference about sytemd GPT partition automounting

https://wiki.archlinux.org/title/Systemd#GPT_partition_automounting

File: alis-commons.conf (hunk 1 above)
Comment (heroldtech) [reply alis-commons.conf:19]: The main reason can be found on https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html.

There it is stated that "When systemd is running in the initrd the / partition may be encrypted in LUKS format as well. In this case, a device mapper device is set up under the name /dev/mapper/root".

File: alis.sh (hunk 2 above)
Comment (heroldtech) [reply alis.sh:427]: Otherwise systemd-gpt-auto-generator is not trying to decrypt the partition.

I am not entirely sure but as far as I understood, according to "The Discoverable Partitions Specification" (see https://uapi-group.org/specifications/specs/discoverable_partitions_specification/), the underlying partition GUID after decryption needs to be specified, which here is Linux x86-64 root (/).

File: alis.sh (hunk 3 above)
Comment (heroldtech) [reply alis.sh:759]: Thinking a second time about it: Only if the root partition is encrypted.

According to https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html: "When systemd is running in the initrd the / partition may be encrypted in LUKS format as well." 
Therefore, if the root partition is LUKS encrypted. it is necessary that, instead of busybox init, systemd init is employed.

I forgot to add the check whether the root partition should be encrypted.

**Comment by heroldtech** [comment]
Regarding GPT partition automounting and encrypted root partition, see also https://wiki.archlinux.org/title/Dm-crypt/Encrypting_an_entire_system#Simple_encrypted_root_with_TPM2_and_Secure_Boot and https://swsnr.de/2022/01/06/install-arch-with-secure-boot-tpm2-based-luks-encryption-and-systemd-homed.html for reference.

File: alis.sh (hunk 3 above)
Comment (heroldtech) [reply alis.sh]: I have updated my branch correspondingly.

File: alis.sh [hunk 4]
```diff
@@ -753,6 +762,10 @@ function mkinitcpio_configuration() {
         if [ -n "$LUKS_PASSWORD" ]; then
             HOOKS=${HOOKS//!sd-encrypt/sd-encrypt}
         fi
+    elif [ "$GPT_AUTOMOUNT" == "true" ] && [ -n "$LUKS_PASSWORD" ]; then
+        HOOKS=${HOOKS//!systemd/systemd}
+        HOOKS=${HOOKS//!sd-vconsole/sd-vconsole}
+        HOOKS=${HOOKS//!sd-encrypt/sd-encrypt}
```
Comment (picodotdev) [inline alis.sh:768]: If BOOTLOADER is not systemd, why systemd hooks area activated?

Comment (heroldtech) [reply alis.sh:768]: From what I understand from https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html, if the root partition is encrypted GPT partition automounting requires systemd init. This then translates to using systemd hooks, see e.g. https://wiki.archlinux.org/title/Mkinitcpio#Common_hooks.

Comment (picodotdev) [reply alis.sh:768]: One thing is the init system (in alis/arch is always systemd) and other is the bootloader (in alis can be systemd-boot, grub or refind).

So, if GPT partition automounting requires systemd init the condition already meets as systemd init is the init used by default in alis/arch. So to me seems this changes are not necessary?

Seems good to you?

Comment (heroldtech) [reply alis.sh:768]: Hmm.. from practice I can say that the changes were necessary. I am not sure if we are meaning the same thing when talking about the init system. 

Therefore I try to be more specific:
From what I understand from "When systemd is running in the initrd the / partition may be encrypted in LUKS format as well." (https://www.freedesktop.org/software/systemd/man/systemd-gpt-auto-generator.html), is that systemd needs to be included in the initramfs (https://wiki.archlinux.org/title/Arch_boot_process#initramfs). This, to me means „systemd init“ and need to be used instead of busybox init, as determined by the mkinitcpio hooks (https://wiki.archlinux.org/title/Mkinitcpio#Common_hooks).

Comment (heroldtech) [reply alis.sh:768]: Systemd init/busybox init are also referred to as early init systems.
